istioctl/cmd/kubeinject.go

```diff
@@ -391,26 +394,65 @@ func setupKubeInjectParameters(sidecarTemplate *inject.Templates, valuesConfig *
 		}
 		return injector, nil, nil
 	}
+
+	// Get configs from IOP files firstly, and if not exists, get configs from files and configmaps.
 	var meshConfig *meshconfig.MeshConfig
-	if meshConfigFile != "" {
-		if meshConfig, err = mesh.ReadMeshConfig(meshConfigFile); err != nil {
-			return nil, nil, err
+	meshConfig, err = setConfigsFromIOPIfNotPresent(valuesConfig, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+	if meshConfig == nil {
+		if meshConfigFile != "" {
+			if meshConfig, err = mesh.ReadMeshConfig(meshConfigFile); err != nil {
+				return nil, nil, err
+			}
+		} else {
+			if meshConfig, err = getMeshConfigFromConfigMap(kubeconfig, "kube-inject", revision); err != nil {
+				return nil, nil, err
+			}
 		}
-	} else {
-		if meshConfig, err = getMeshConfigFromConfigMap(kubeconfig, "kube-inject", revision); err != nil {
+	}
+	if valuesConfig == nil || *valuesConfig == "" {
+		if valuesFile != "" {
+			valuesConfigBytes, err := os.ReadFile(valuesFile) // nolint: vetshadow
+			if err != nil {
+				return nil, nil, err
+			}
+			*valuesConfig = string(valuesConfigBytes)
+		} else if *valuesConfig, err = getValuesFromConfigMap(kubeconfig, revision); err != nil {
 			return nil, nil, err
 		}
 	}
-	if valuesFile != "" {
-		valuesConfigBytes, err := os.ReadFile(valuesFile) // nolint: vetshadow
+	return injector, meshConfig, err
+}
+
+// setConfigsFromIOPIfNotPresent sets the configs with configs in IOPs.
+func setConfigsFromIOPIfNotPresent(valuesConfig *string, logger clog.Logger) (*meshconfig.MeshConfig, error) {
+	var meshConfig *meshconfig.MeshConfig
+	if iopFilename != "" {
+		_, iops, err := manifest.GenerateConfig([]string{iopFilename}, nil, false, nil, logger)
```

Comment:
GenerateConfig seems like its doing way more than what we want. Can we not just unmarshal it ?

Reply:
Done.

Reply:
Done.